src/fastq/call_consensus_reads/pipeline.rs

```diff
@@ -419,36 +463,46 @@ impl<'a, R: io::Read, W: io::Write> CallConsensusReads<'a, R, W>
                 median_distances.push((median_hamming_distance, insert_size))
             }
         }
-
         //TODO Add deterministic uuid considering read ids
         let uuid = &Uuid::new_v4().to_hyphenated().to_string();
-        if let Some(consensus_record) = median_distances
+        //Calculate most likely consensus record
+        let (consensus_record, ol_lh_isize, ol_consensus_isize) = median_distances
             .iter()
-            .filter_map(|(mean_distance, insert_size)| {
-                if *mean_distance < HAMMING_THRESHOLD {
+            .filter_map(|(median_distance, insert_size)| {
+                if *median_distance < HAMMING_THRESHOLD {
                     let overlap = (f_recs[0].seq().len() + r_recs[0].seq().len()) - insert_size;
-
-                    Some(
-                        CalcOverlappingConsensus::new(
-                            &f_recs,
-                            &r_recs,
-                            overlap,
-                            &outer_seqids,
-                            &uuid,
-                        )
-                        .calc_consensus(),
+                    let (consensus_record, lh) = CalcOverlappingConsensus::new(
+                        &f_recs,
+                        &r_recs,
+                        overlap,
+                        &outer_seqids,
+                        &uuid,
                     )
+                    .calc_consensus();
+                    Some((consensus_record, lh, *insert_size))
```

Comment:
Return a struct here. Put it in the variable `consensus_overlap`. Return a struct below, put it in the variable `consensus_no_overlap`. THe struct should be like this:
```
pub struct Consensus {
    record: ...,
    likelihood: ...,
    insert_size: ...
}
```

Thereby likelihood is the likelihood of the consensus sequence, i.e., Pr(R, I | C) = Pr(R | C, I) * Pr(I).